docs/en/relying-party-solution.rst

```diff
@@ -139,6 +132,22 @@ Below follows its Base64 decoded content:
   eudiw://authorize?client_id=https://verifier.example.org&request_uri=https://verifier.example.org/request_uri
 
 
+Same Device Flow
+----------------
+In the **Same Device Authorization Flow**, the User interacts with a Verifier that resides in the same device of the Wallet Instance.
+This scenario utilizes HTTP redirects to finalize the authorization phase and obtain Verifiable Presentation(s).
+
+The flow is identical to the cross device with the only replacement of the steps relating to the use of the QRCode, i.e. steps number **3**, **4**, **5** and **6**, which are replaced by a single step described below.
+
+The Relying Party produces a URL similar to the one contained in the QRCode produced in the Cross Device Flow, then provides it to the Wallet Instance via an HTTP 302 response (redirect), represented in the following non-normative example:
+
+.. code:: text
+
+    HTTP/1.1 /pre-authz-endpoint Found
+    Location: https://verifier.example.org/request_uri_endpoint?
+    client_id=https%3A%2F%2Fverifier.example.org%2Fcb
+    &request_uri=https%3A%2F%2Fverifier.example.org%2Frequest_uri_endpoint
+
```

Comment:
Shall we add that the user will first have to select their wallet provider?

Reply:
Why?

in the presentation phase the user interact directly to the RP with their wallet instance. 

What would you like to say?

Reply:
Please, we really have to take out the possibility that the user should select the wallet provider in a RP discovery page.

The RP provides the QRCode, the User open their Wallet Instance and starts the flow.
At the end of the road we must adopt all the compromises to have the EUDI interop, then every member state will support the same things (even with custom extension but the same for interop needs)

for the PoC the user knows that a specific app is needed to start the flow